inc/HashMap.h

```diff
@@ -18,76 +20,66 @@ namespace CTSL //Concurrent Thread Safe Library
     //Each hash bucket is implemented as singly linked list with the head as a dummy node created 
     //during the creation of the bucket. All the hash buckets are created during the construction of the map.
     //Locks are taken per bucket, hence multiple threads can write simultaneously in different buckets in the hash map
-    template <typename K, typename V, typename F = std::hash<K> >
+    template <typename K, typename V, typename HASH = std::hash<K> >
     class HashMap
     {
         public:
             HashMap(size_t hashSize_ = HASH_SIZE_DEFAULT) : hashSize(hashSize_)
             {
-                hashTable = new HashBucket<K, V> * [hashSize](); //create the hash table as an array of hash buckets
-                for(size_t i = 0; i < hashSize; i++)
-                {
-                    hashTable[i] = new HashBucket<K, V>(); //create the hash buckets
-                }
+                assert(hashSize_ > 0 && "Hash size muste be > 0");
+                assert((hashSize_ & (hashSize_ - 1)) == 0 && "Hash size must be power of 2");
```

Comment:
Seriously, if you're going to change the API contract, you really must update the documentation.

Reply:
What API constract and documentation are you talking about?